add Phase2 workflows to PR tests

kpedro88 opened the pull request:
Choice of workflows/processes to add:
- 2023LReco has HGCal
- 2023GReco has full reco workflow (currently with Run2 calorimeters)
- ttbar tends to exercise the most algorithms

attn: @boudoul, @calabria, @slava77

kpedro88 committed:
add Phase2 workflows to PR tests

cmsbuild commented:
A new Pull Request was created by  @kpedro88 (Kevin Pedro) for CMSSW_8_1_X.

It involves the following packages:

Configuration/PyReleaseValidation


@cmsbuild, @srimanob, @davidlange6, @hengne, @fabozzi can you please review it and eventually sign? Thanks.
@ghellwig, @Martin-Grunewald this is something you requested to watch as well.
@slava77, @degano, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

kpedro88 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 431714d5c6a64ee7475be76f8b19459ce0110374
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14977/13674/summary.html

slava77 commented on Configuration/PyReleaseValidation/scripts/runTheMatrix.py:
```diff
@@ -56,7 +56,9 @@ def runSelected(opt):
-                     135.4 #Run 2 Zee ttbar
+                     135.4, #Run 2 Zee ttbar
+                     11024.0, #2023LReco ttbar (w/ HGCal)
+                     11224.0  #2023GReco ttbar (flat tracker, Run2 calo)
```
need to decide if we want it here (for everyone running the short matrix) or in 
https://github.com/cms-sw/cms-bot/blob/master/cmssw-pr-test-config for jenkins jobs
Note that 2017 wflows are only in jenkins extras (10021.0,10024.0 )
If here, it makes more sense then to put 10021.0 and 10024.0  here as well

kpedro88 replied:
It's fine with me to put it only in the bot's config (I didn't realize it had a separate config for PR tests). But since we sometimes use -l limited to test PRs privately, it could be nice to have them there. I would probably prefer to add 10021.0 and 10024.0 to limited.

kpedro88 replied:
@slava77 @davidlange6 we should make a decision here - is it okay to put Phase2 (and additionally 2017?) in the limited matrix, or should I update just the bot's list of extra PR tests?

slava77 replied:
I was ruminating on this.
I think it's better to add both the pair of 2023 and the 2017 here in the short matrix explicitly.
These are reasonably short workflows (as long as you can submit them all in parallel).

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14977/13674/summary.html

kpedro88 committed:
add 2017 WFs to limited

cmsbuild commented:
Pull request #14977 was updated. @cmsbuild, @srimanob, @davidlange6, @hengne, @fabozzi can you please check and sign again.

kpedro88 commented:
@slava77 I added the 2017 WFs to limited. I also switched the full reco workflow to tilted tracker, since Meenakshi said we should treat that as the "default" option for now.

kpedro88 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

kpedro88 commented:
@slava77 also, will we need a separate PR for the bot to remove the (now duplicated) 2017 WFs from the test config?

cmsbuild commented:
+1
Tested at: f9f79225aab258ce2d86f0fceb087f0c532bdf90
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14977/13742/summary.html

slava77 commented:
On 6/28/16 6:25 PM, Kevin Pedro wrote:
> @slava77 <https://github.com/slava77> also, will we need a separate PR
> for the bot to remove the (now duplicated) 2017 WFs from the test config?

Yes, please.

I don't think this is required though, but better do it.


> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/14977#issuecomment-229103363>, or
> mute the thread
> <https://github.com/notifications/unsubscribe/AEdcboSdjpafZmH-hNqj9TjMZyaFwMMuks5qQUsDgaJpZM4I-RVf>.
>

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14977/13742/summary.html

kpedro88 commented:
@slava77 - submitted https://github.com/cms-sw/cms-bot/pull/720

@hengne - please take a look when you can...

kpedro88 commented:
@hengne ping

fabozzi commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_1_X IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @degano, @smuzaffar

kpedro88 commented:
@davidlange6 would like to merge this ASAP to reduce chances of the workflows breaking before the tests are there...

davidlange6 commented:
+1

davidlt commented:
These workflows tend to ask for non-existing datasets as input, e.g. `/RelValTenMuE_0_200/1/GEN-SIM` in 11021.0. There are a number of such workflows asking for non-existing input datasets in IB RelVals. Could you clean that up?

davidlt commented:
/cc @smuzaffar

davidlt commented:
From `Configuration/PyReleaseValidation/python/relval_steps.py`

```
1564 #################################################################################
1565 ####From this line till the end of the file :
1566 ####UPGRADE WORKFLOWS IN PREPARATION - Gaelle's sandbox -
1567 #####Accessible only through the option --what upgrade
1568 #####therefore not run in IBs (at some they might be...)
1569 #####Transparent for any of the standard workflows
1570 #### list of worflows defined (not necessarly running though): runTheMatrix.py --what upgrade -n
1571 ####
1572 ###
1573 #################################################################################
```
Is this still true? Because according to comment this probably shouldn't be run in IB RelVals.

slava77 commented:
@davidlt were you checking with ``` -i all ``` ?
11021.0 works only with resim (full step 1) at this point.
standard jenkins tests use resim.

@kpedro88 we should either move this workflow to jenkins-only list of extras or you'd need to define a valid GEN-SIM input.
... or we live with the problem in ``` -i all ```  setup.

kpedro88 commented:
@davidlt I'm not sure where that is specified. The only result for that dataset name in GitHub search is: https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/Validation/RecoTrack/test/trackingPerformanceValidation.py

@makortel?

kpedro88 commented:
@davidlt I made the workflows appear in the regular matrix in this PR.

davidlt commented:
@slava77 yes, you can find these workflows is standard IB RelVals. O noticed those as I was yesterday checking DAS errors to request data back to T2_CH_CERN.

@kpedro88 if you have enabled this for standard IB RelVals, then you have yo clean up that comment. Plus run GEN-SIM for these workflows.

I have a patch somewhere stashed which would check for dataset in DBS and if not found switch from recycling to GEN-SIM mode, but I never published it.

davidlt commented:
IBs are always running with `-i all`.

makortel commented:
@kpedro88 https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/Validation/RecoTrack/test/trackingPerformanceValidation.py has nothing to do with standard jenkins/IB/RelVal production.

slava77 commented:
On 7/13/16 6:37 AM, davidlt wrote:
> IBs are always running with |-i all|.
> 

to run in jenkins with all other PR test workflows, we need this one in
the standard list (not -w upgrade).
Default jenkins tests run resim
and they should keep having this workflow 11021 included.

I'd suggest to apply a patch to run resim if inputs are not available in
 "-i all" option
or just live with the failures in IBs.


> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/14977#issuecomment-232357001>, or
> mute the thread
> <https://github.com/notifications/unsubscribe/AEdcbkcsYVYxiwgWMHrWKtWtGNWhSc5oks5qVOoKgaJpZM4I-RVf>.
>

kpedro88 commented:
@davidlt I honestly don't know where that GEN-SIM dataset gets specified. I opened issue #14975 before making this PR to address these kinds of issues and never heard about it. If you know where it gets set, please point it out.

davidlt commented:
IIRC, if `-i all` is specified it looks for `step0 + INPUT`, otherwise for `step0`. Looks like you are building these workflows dynamically and it's messy.

E.g.,
```
$ runTheMatrix.py -i all -n | grep 11021
11021.0 TenMuE_0_200_pythia8_2023LReco_GenSimFullINPUT+DigiFull_2023LReco+RecoFullLocal_2023LReco
```

You can see that the first step has suffix `INPUT` thus it will look for data from whatever configured site (in our case T2_CH_CERN).
Normally one would need here `TenMuE_0_200_pythia8_2023LReco_GenSimFull` which would define a GEM-SIN `cmsDriver.py` command.

@slava77 we don't have non-working/failing workflows or test within our test suite, otherwise not will care to look at the results.

kpedro88 commented:
For Phase2, the geometry and simulation code are changing frequently. We really need to have the GEN-SIM generated on the fly for these workflows whenever they are used for tests. Is there some way to prevent the `INPUT` suffix from being applied to the Phase2 workflows?

davidlt commented:
I have no idea how it auto-generated step0 INPUT step. One needs to add a few print statements to debug it. Then see if that can be changed to generate GEN-SIM step0 instead of INPUT stuff.

kpedro88 commented:
@davidlt Okay, I'll try to debug it.

slava77 commented:
On 7/13/16 7:03 AM, davidlt wrote:

> @slava77 <https://github.com/slava77> we don't have non-working/failing
> workflows or test within our test suite, otherwise not will care to look
> at the results.

Where is that "we don't have non-working workflows"?
I haven't seen  IBs regularly with all relval tests complete for a few
years, if ever.
Maybe I misunderstand the scope


> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/14977#issuecomment-232364759>, or
> mute the thread
> <https://github.com/notifications/unsubscribe/AEdcbnnmWDgtm8I1CvMGFlwxd8F5Fttaks5qVPAegaJpZM4I-RVf>.
>

slava77 commented:
On 7/13/16 6:50 AM, Kevin Pedro wrote:
> @davidlt <https://github.com/davidlt> I honestly don't know where that
> GEN-SIM dataset gets specified.

Kevin,
 you know it, it's your commit that put dummy values
https://github.com/cms-sw/cmssw/pull/14865/commits/bbc26dc46be56e54bb16c0fd37ff8f713ab82b69


>  I opened issue #14975
> <https://github.com/cms-sw/cmssw/issues/14975> before making this PR to
> address these kinds of issues and never heard about it. If you know
> where it gets set, please point it out.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/14977#issuecomment-232360792>, or
> mute the thread
> <https://github.com/notifications/unsubscribe/AEdcbgRiGPYVQv5y2C_fmsuBuXdEM3Eoks5qVO03gaJpZM4I-RVf>.
>

kpedro88 commented:
@slava77 that just sets the *minbias* sample for the 2023 PU workflows to an empty string. @davidlt sees something different, a non-empty string pointing to a non-existing sample. This must be automatically generated somewhere in PyReleaseValidation; I will try to figure out where.

slava77 commented:
On 7/13/16 7:56 AM, Kevin Pedro wrote:
> @slava77 <https://github.com/slava77> that just sets the /minbias/
> sample for the 2023 PU workflows to an empty string. @davidlt
> <https://github.com/davidlt> sees something different, a non-empty
> string pointing to a non-existing sample. This must be automatically
> generated somewhere in PyReleaseValidation; I will try to figure out where.

I think this controls both
try to revert and see.
https://github.com/kpedro88/cmssw/blob/bbc26dc46be56e54bb16c0fd37ff8f713ab82b69/Configuration/PyReleaseValidation/python/relval_steps.py#L1591-L1597


> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/14977#issuecomment-232381653>, or
> mute the thread
> <https://github.com/notifications/unsubscribe/AEdcbjnYCJw7gzki-p-i29vK3sbM5FoLks5qVPy6gaJpZM4I-RVf>.
> 

\

kpedro88 commented:
@slava77 @davidlt 

It's actually here: https://github.com/kpedro88/cmssw/blob/bbc26dc46be56e54bb16c0fd37ff8f713ab82b69/Configuration/PyReleaseValidation/python/relval_steps.py#L1788-L1799

How should we fix it? Just removing the InputInfo assignment seems like it will still break things...

kpedro88 commented:
@slava77 I guess we were both right, since the above-linked lines append the value from `baseDataSetReleaseBetter`. I still think we prefer to keep that as an empty string.

davidlt commented:
A quick look and yes, that seems to the the place to generate step0 INPUT.

This is `if 'Sim' in step:` Shouldn't it then generate GEN-SIM cmsDriver command? Do you have a similar blocks which generates GEN-SIM cmsDriver commands?

https://github.com/kpedro88/cmssw/blob/bbc26dc46be56e54bb16c0fd37ff8f713ab82b69/Configuration/PyReleaseValidation/python/relval_steps.py#L1590

Here we get that "1" (version) which is later put in the dataset.

kpedro88 commented:
I'm not sure what you mean by "a block which generates GEN-SIM cmsDriver commands". That is the default operation of runTheMatrix, isn't it? It's only with this special `-i` option that we have to deal with the INPUT issue.

kpedro88 commented:
@davidlt @slava77 the code appears to be smarter than I thought. I commented out these two lines:
https://github.com/kpedro88/cmssw/blob/bbc26dc46be56e54bb16c0fd37ff8f713ab82b69/Configuration/PyReleaseValidation/python/relval_steps.py#L1798-L1799
and then runTheMatrix no longer tries to reuse input, even with the `-i` flag. Is this a satisfactory solution for everyone?

slava77 commented:
On 7/13/16 9:04 AM, Kevin Pedro wrote:
> @davidlt <https://github.com/davidlt> @slava77
> <https://github.com/slava77> the code appears to be smarter than I
> thought. I commented out these two lines:
> https://github.com/kpedro88/cmssw/blob/bbc26dc46be56e54bb16c0fd37ff8f713ab82b69/Configuration/PyReleaseValidation/python/relval_steps.py#L1798-L1799
> and then runTheMatrix no longer tries to reuse input, even with the |-i|
> flag. Is this a satisfactory solution for everyone?
> 

I can't see clearly if this affects only the 2023 workflows in question.
2017 and others should use "-i all" consistently.


> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/14977#issuecomment-232403301>, or
> mute the thread
> <https://github.com/notifications/unsubscribe/AEdcbp5TLViMOI1lHP1DuoydHZwX1yUHks5qVQyYgaJpZM4I-RVf>.
> 

\

kpedro88 commented:
@slava77 you're right, I'll come up with a condition so it only affects 2023.

kpedro88 commented:
see #15196
